pkg/prometheus/operator.go

```diff
@@ -1589,7 +1589,7 @@ func (c *Operator) createOrUpdateConfigurationSecret(ctx context.Context, p *mon
 		additionalAlertRelabelConfigs,
 		additionalAlertManagerConfigs,
 		ruleConfigMapNames,
-	)
+		store)
```

Comment:
to be consistent with basic auth, oauth and bearer tokens, we shouldn't have to pass the `store` variable to the config generator. Instead `assets.Store` should be extended to support sigv4 secrets and `GenerateConfig()` gets a new `store.Sigv4Assets` argument. Hope it makes sense :)

Reply:
That is kind of what I mentioned in https://github.com/prometheus-operator/prometheus-operator/pull/3994#issuecomment-872129578.

So in the intended solution, `store` will only be used to fetch a new `additionalSigV4Configs` as a `[]byte`? This would align the sigV4 config to others similar parameters passed?

Reply:
Ah sorry i missed your initial comment! Currently the operator will collect the secret materials before synthesizing the configuration because it allows to skip service/pod monitors and probes that have invalid key selectors. Otherwise a "bad" resource could fail the reconciliation process.

So yes, the idea is that the `assets.Store` struct gets 2 new fields: 
* `SigV4AccessKeyAssets map[string]SigV4AccessKey` 
* `SigV4SecretKeyAssets map[string]SigV4SecretKey` 

https://github.com/prometheus-operator/prometheus-operator/blob/3f047bc9c74105ba774aff726798a66b81409f7f/pkg/assets/store.go#L40-L49

You need also 2 new methods (similar to AddBearerToken for instance):
* `AddSigV4AccessKey(ctx context.Context, ns string, sel v1.SecretKeySelector, key string) error` 
* `AddSigV4SecretKey(ctx context.Context, ns string, sel v1.SecretKeySelector, key string) error`

https://github.com/prometheus-operator/prometheus-operator/blob/3f047bc9c74105ba774aff726798a66b81409f7f/pkg/assets/store.go#L204-L218

And then call these functions whenever we do for `AddBearerToken` and others, for instance:

https://github.com/prometheus-operator/prometheus-operator/blob/3f047bc9c74105ba774aff726798a66b81409f7f/pkg/prometheus/operator.go#L1743-L1759

And the configuration generator function should be extended with 2 new arguments: `sigV4AccessKeyAssets map[string]store.SigV4AccessKey` and `sigV4SecretKeyAssets map[string]store.SigV4SecretKey`.

https://github.com/prometheus-operator/prometheus-operator/blob/3f047bc9c74105ba774aff726798a66b81409f7f/pkg/prometheus/promcfg.go#L159-L171

Does it make sense?

Side-note: #4180 refactors and simplifies the `GenerateConfig()` part in a nice way but it's not there yet so feel free to stick with what exists today.